docs/src/reviewing_guide.md

```diff
@@ -0,0 +1,95 @@
+# Reviewing Guide
+
+## Overview
+The purpose of this document is to help reviewers look for common mistakes and vulnerabilities in the code base. This aims
+to help upskill new developers and reviewers to the project. It is by no means a complete list of things to look for, but
+it should help to get started.
+
+This is a living document and should be updated as new issues are found and new things are learned.
+
+## Reviewing Process
+
+Many vulnerabilities start at the edges of the system, for example, an attacker may craft a data packet that causes a panic.
+So a good place to start is to look at the entry points to the system and follow the packet through the flow of the system. 
+This is generally much more effective than trying to look at a file in isolation. 
+
+### Trusted vs Untrusted Data
+Data received that is not gener
+ated by the local system should be seen as untrusted and treated very carefully.
+
+In addition to data packets received from the network, there are other sources of untrusted data. For example, a config 
+setting may be copied from a post on the internet, or command line interface (CLI) commands may be copied from a forum.
+
+Even the local data files that are generated in the data directory may be untrusted in some scenarios. For blockchains, it 
+is not uncommon to download a copy of the blockchain from a third party as a data file. 
+
+Another, less obvious, source of untrusted data are third party crates. Changes to dependencies should be monitored closely.
+
+So what are some things to look out for when dealing with untrusted data?
+1. Any parsing of untrusted data can panic, return an error or generate the wrong data. If the code parsing is inside the
+Tari codebase, it should have matching tests and be fuzzed. If the code is in a third party crate, it should be reviewed very
+carefully. Does the third party crate have fuzzing? Does it have tests? Is it actively maintained? 
+    1. Be careful when using crates that wrap C libraries and other native code. A panic in native code can not be recovered 
+and could be used to crash a node.
+2. Be especially careful when reading the length for a buffer or Vec from untrusted data. If the length is too large, it could
```

Comment:
I would branch this out, and more be care when working with buffers. 
Point 1: length trusting in terms of buffer overflow
Point 2: length trusting in terms of sending extra data. 
eg: vec<u8> = [1,2,3,4,5,6], reported len is 2, data might be deserialized as vex<u8> = [1,2] which is the correct object, but the packet still carries the data [3,4,5,6]

Reply:
added below